internal/hub/hubclient/hubclient.go

```diff
@@ -146,6 +147,132 @@ func updateClusterInfoToHub(ctx context.Context, spokeclient client.Client, hubC
+	} else if nsIndex == -1 {
+		hubCluster.Status.Namespaces = append(hubCluster.Status.Namespaces, hubv1alpha1.NamespacesConfig{
+			Name:      onboardNamespace,
+			SliceName: sliceName,
+		})
+		log.Info("Adding namespace on hub cluster", "cluster", ClusterName, "namespace", onboardNamespace)
+		return updateNamespaceInfoToHub(hubClient, ctx, hubCluster, onboardNamespace, sliceName)
+	}
+	return nil
+}
+func updateNamespaceInfoToHub(hubClient client.Client, ctx context.Context,
+	hubCluster *hubv1alpha1.Cluster, onboardNamespace, sliceName string) error {
+	if err := hubClient.Status().Update(ctx, hubCluster); err != nil {
+		if errors.IsConflict(err) {
+			log.Error(err, "conflict error", "namespace", onboardNamespace)
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				err := hubClient.Get(ctx, types.NamespacedName{
+					Name:      os.Getenv("CLUSTER_NAME"),
+					Namespace: os.Getenv("HUB_PROJECT_NAMESPACE"),
+				}, hubCluster)
+				if err != nil {
+					return err
+				}
+				hubCluster.Status.Namespaces = append(hubCluster.Status.Namespaces, hubv1alpha1.NamespacesConfig{
+					Name:      onboardNamespace,
+					SliceName: sliceName,
+				})
+				log.Info(" Retrying adding namespace on hub cluster", "cluster", os.Getenv("CLUSTER_NAME"), "namespace", onboardNamespace)
+				return hubClient.Status().Update(ctx, hubCluster)
+			})
+			return err
+		}
+		return err
+	}
+	log.Info("Successfully update cluster namespace", "namespace", onboardNamespace)
+	return nil
+}
+
+// gets the index of worker namespace from hub cluster CR array
+func indexOf(onboardNamespace string, ns []hubv1alpha1.NamespacesConfig) int {
+	for k, v := range ns {
+		if onboardNamespace == v.Name {
+			return k
+		}
+	}
+	return -1
+}
+
+func DeleteNamespaceInfoFromHub(ctx context.Context, hubClient client.Client, onboardNamespace string) error {
+	hubCluster := &hubv1alpha1.Cluster{}
+	err := hubClient.Get(ctx, types.NamespacedName{
+		Name:      os.Getenv("CLUSTER_NAME"),
+		Namespace: os.Getenv("HUB_PROJECT_NAMESPACE"),
+	}, hubCluster)
+	if err != nil {
+		return err
+	}
+	toDeleteNs := indexOf(onboardNamespace, hubCluster.Status.Namespaces)
+	if toDeleteNs == -1 {
+		return nil
+	}
+	hubCluster.Status.Namespaces = append(hubCluster.Status.Namespaces[:toDeleteNs],
+		hubCluster.Status.Namespaces[toDeleteNs+1:]...)
+	log.Info("Deleting namespace on hub cluster", "cluster", ClusterName, "namespace", onboardNamespace)
+	return deleteNamespaceInfoToHub(hubClient, ctx, hubCluster, onboardNamespace)
+}
+
+func deleteNamespaceInfoToHub(hubClient client.Client, ctx context.Context,
+	hubCluster *hubv1alpha1.Cluster, onboardNamespace string) error {
+	if err := hubClient.Status().Update(ctx, hubCluster); err != nil {
+		if errors.IsConflict(err) {
+			log.Error(err, "conflict error", "namespace", onboardNamespace)
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Can we use retry.RetryOnConflict() directly instead of using it only if the first update attempt fails? It would look much cleaner that way.

Reply:
Sure let me update it